docs/src/main/paradox/scala/http/server-side/http2.md

```diff
@@ -0,0 +1,83 @@
+# HTTP/2 (Preview)
+
+HTTP/2 support in akka-http is currently available as a preview.
+This means it is ready to be evaluated, but the API's and behavior is likely to change.
+
+@@@ div { .group-scala }
+
+## Enabling HTTP/2
+
+sbt
+:   @@@vars
+    ```sbt
+    "com.typesafe.akka" %% "akka-http2-support" % "$project.version$"
+    ```
+    @@@
+
+maven
+:   @@@vars
+    ```
+    <dependency>
+        <groupId>com.typesafe.akka</groupId>
+        <artifactId>akka-http2-support_2.12</artifactId>
+        <version>$project.version$</version>
+    </dependency>
+    ```
+    @@@
+
+We only support HTTP/2 over HTTPS: browsers don't typically support HTTP/2 on plain HTTP, and other clients that do are rare.
+
+To explicitly bind a server that accepts both HTTP and HTTP/2, simply use @scaladoc[Http2().bindAndHandleAsync](akka.http.scaladsl.Http2Ext) in the same way you would @scaladoc[Http().bindAndHandleAsync](akka.http.scaladsl.HttpExt):
+
+Scala
+:   @@snip[Http2Spec.scala](../../../../../test/scala/docs/http/scaladsl/Http2Spec.scala) { #bindAndHandleAsync }
+
+Java
+:   @@snip[Http2Test.java](../../../../../test/java/docs/http/javadsl/Http2Test.java) { #bindAndHandleAsync }
+
+For an even more transparent experience, you can use the regular @scaladoc[Http().bindAndHandleAsync](akka.http.scaladsl.HttpExt) and enable HTTP/2 through configuration, but make sure you include the dependency and use the `Async` variation of `bindAndHandle`:
+
+```
+akka.http.server.preview.enable-http2 = on
+```
+
+At this point you should be able to connect, which you can verify with `curl -v`:
```

Comment:
I didn't do anything special for it:
```
$ curl --version
curl 7.52.1 (x86_64-pc-linux-gnu) libcurl/7.52.1 OpenSSL/1.0.2l zlib/1.2.8 libidn2/0.16 libpsl/0.17.0 (+libidn2/0.16) libssh2/1.8.0 nghttp2/1.23.1 librtmp/2.3
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: AsynchDNS IDN IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz TLS-SRP HTTP2 UnixSockets HTTPS-proxy PSL
```
.. but indeed the version in Debian Jessie doesn't seem to include it. We should probably mention you need a reasonably recent version of curl.